src/gpu/intel/ocl/micro_sdpa.cl

```diff
@@ -225,11 +225,16 @@ DECLARE_2D_TILE_RSELECT(a_scale_tile_type, SUBGROUP_SIZE, ugemm_vs_sg_tile_n, 1,
      for (int i = 0; i < 8; i++)
          f = A_tile1.x[i][0];
 */
+
 __attribute__((intel_reqd_sub_group_size(SUBGROUP_SIZE))) kernel void
 micro_sdpa(const global KEY_DATA_T *K, const global QRY_DATA_T *Q,
         const global VAL_DATA_T *V, global DST_DATA_T *A,
-        const global SCALE_DATA_T *scale_ptr, int d, int k, int q,
-        const global KEY_ATTR_SCALES_DATA_T *K_scales,
+#if HOST_SIDE_SCALE
+        const float scale, const float iscale,
+#else
+        const global SCALE_DATA_T *scale_ptr,
+#endif
+        int d, int k, int q, const global KEY_ATTR_SCALES_DATA_T *K_scales,
```

Comment:
DISCUSSION: There may be an issue with OpenVINO dealing with dual macros in this kernel signature. It has been suggested that we may be able to deal with the issue by relying on an USM approach: a pointer is passed to memory containing the scalar. 

On the other hand, this approach reduces compute inside the kernel and keeps typing of `scale` with convention. It also relies on passing a scalar to the kernel through absolutely basic OpenCL reducing any unexpected behavior due to driver errors, synchronization, etc.